cmake/macros/macro_define_interface_target.cmake

```diff
@@ -0,0 +1,148 @@
+#   targets), respectively.
+#
+# The default name, i.e., interface_<feature>(|_debug|_release) can be
+# overriden by the optional second argument. For example,
+#   define_interface_target(DEAL_II base_configuration)
+# will define interface_base_configuration* targets but query all
+# information from DEAL_II_* variables.
+#
+
+function(define_interface_target _feature)
+  string(TOLOWER "${_feature}" _feature_lowercase)
+  if(NOT "${ARGN}" STREQUAL "")
+    set(_feature_lowercase "${ARGN}")
+  endif()
+
+  set(_builds "dummy")
+  if(${_feature}_SPLIT_CONFIGURATION)
+    set(_builds ${DEAL_II_BUILD_TYPES})
+  endif()
+
+  foreach(_build ${_builds})
+    set(_interface_target "interface_${_feature_lowercase}")
+
+    if(${_feature}_SPLIT_CONFIGURATION)
+      string(TOLOWER "${_build}" _build_lowercase)
+      string(APPEND _interface_target "_${_build_lowercase}")
+    endif()
+
+    message(STATUS "")
+    message(STATUS "Defining target: ${_interface_target}")
+
+    add_library(${_interface_target} INTERFACE)
+
+    if(DEFINED ${_feature}_VERSION)
+      message(STATUS "    VERSION:             ${${_feature}_VERSION}")
+      set_target_properties(${_interface_target}
+        PROPERTIES VERSION "${${_feature_}_VERSION}"
+        )
+    endif()
+
+    set(_libraries)
+    list(APPEND _libraries
+      ${${_feature}_LIBRARIES} ${${_feature}_LIBRARIES_${_build}}
+      ${${_feature}_TARGETS} ${${_feature}_TARGETS_${_build}}
+      )
+    if(NOT "${_libraries}" STREQUAL "")
+      message(STATUS "    LINK_LIBRARIES:      ${_libraries}")
+      target_link_libraries(${_interface_target} INTERFACE ${_libraries})
+    endif()
+
+    if(DEFINED ${_feature}_INCLUDE_DIRS)
+      message(STATUS "    INCLUDE_DIRECTORIES: ${${_feature}_INCLUDE_DIRS}")
+      target_include_directories(${_interface_target}
+        SYSTEM INTERFACE ${${_feature}_INCLUDE_DIRS}
+        )
+    endif()
+
+    set(_definitions)
+    list(APPEND _definitions ${${_feature}_DEFINITIONS} ${${_feature}_DEFINITIONS_${_build}})
+    if(NOT "${_definitions}" STREQUAL "")
+      message(STATUS "    COMPILE_DEFINITIONS: ${_definitions}")
+      target_compile_definitions(${_interface_target} INTERFACE ${_definitions})
+    endif()
+
+    separate_arguments(_compile_options UNIX_COMMAND
+      "${${_feature}_CXX_FLAGS} ${${_feature}_CXX_FLAGS_${_build}}"
+      )
+    if(NOT "${_compile_options}" STREQUAL "")
+      message(STATUS "    COMPILE_OPTIONS:     ${_compile_options}")
+      target_compile_options(${_interface_target} INTERFACE $<$<COMPILE_LANGUAGE:CXX>:${_compile_options}>)
+    endif()
+
+    separate_arguments(_link_options UNIX_COMMAND
+      "${${_feature}_LINKER_FLAGS} ${${_feature}_LINKER_FLAGS_${_build}}"
+      )
+    if(NOT "${_link_options}" STREQUAL "")
+      message(STATUS "    LINK_OPTIONS:        ${_link_options}")
+      target_link_options(${_interface_target} INTERFACE ${_link_options})
+    endif()
```

Comment:
I guess this should use generator expressions as well? At least for `Kokkos` we are currently manually extracting the linker flags from an expression that contains generator expressions.

Reply:
Good point. I will guard these with a `COMPILE_LANGUAGE` generator expression as well.

Reply:
Fixed.